components/table/table-wrapper.js

```diff
@@ -0,0 +1,318 @@
+		right: 0;
+	}
+
+	/* first row: offset by size of border-radius so left/right border doesn't show through (default style only) */
+	d2l-table-wrapper[sticky-headers][type="default"] .d2l-table > * > tr[header] > *,
+	d2l-table-wrapper[sticky-headers][type="default"] .d2l-table > thead > tr > th {
+		top: -5px;
+	}
+
+	/* first column that's sticky: offset by size of border-radius so top/bottom border doesn't show through (default style only) */
+	d2l-table-wrapper[sticky-headers][type="default"]:not([dir="rtl"]) .d2l-table > * > tr > [sticky].d2l-table-cell-first {
+		left: -5px;
+	}
+	d2l-table-wrapper[sticky-headers][type="default"][dir="rtl"] .d2l-table > * > tr > [sticky].d2l-table-cell-first {
+		right: -5px;
+	}
+
+	/* non-header sticky cells */
+	d2l-table-wrapper[sticky-headers] .d2l-table > * > tr:not([selected]) {
+		background-color: inherit; /* white background so sticky cells layer on top of non-sticky cells */
+	}
+	d2l-table-wrapper[sticky-headers] .d2l-table > tbody > tr:not([header]) > [sticky] {
+		background-color: inherit;
+		position: -webkit-sticky;
+		position: sticky;
+		left: 0;
+		z-index: 1;
+	}
+`;
+
+/**
+ *
+ * Wraps a native <table> element, providing styling and scroll buttons for overflow.
+ *
+ * @slot - Content to wrap
+ */
+export class TableWrapper extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether header row is sticky
+			 */
+			stickyHeaders: {
+				attribute: 'sticky-headers',
+				reflect: true,
+				type: Boolean
+			},
+			/**
+			 * Type of table style to apply
+			 * @type {'default'|'light'}
+			 */
+			type: {
+				reflect: true,
+				type: String
+			}
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+				width: 100%;
+				--d2l-table-border: 1px solid var(--d2l-table-border-color);
+				--d2l-table-border-color: var(--d2l-color-mica);
+				--d2l-table-border-radius: 0.3rem;
+				--d2l-table-cell-height: 41px; /* min-height to be 62px including border */
+				--d2l-table-cell-padding: 0.5rem 1rem;
+				--d2l-table-header-background-color: var(--d2l-color-regolith);
+				--d2l-table-row-border-color-selected: var(--d2l-color-celestine);
+				--d2l-table-row-background-color-selected: var(--d2l-color-celestine-plus-2);
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			:host([type="light"]) {
+				--d2l-table-cell-height: 1.15rem; /* min-height to be 48px including border */
+				--d2l-table-cell-padding: 0.6rem;
+				--d2l-table-border-color: var(--d2l-color-gypsum);
+				--d2l-table-header-background-color: #ffffff;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.stickyHeaders = false;
+		this.type = 'default';
+		this._tableObserver = null;
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._tableObserver) this._tableObserver.disconnect();
+	}
+
+	render() {
+		const slot = html`<slot @slotchange="${this._handleSlotChange}"></slot>`;
+		if (this.stickyHeaders) {
```

Comment:
Previously it would tell scroll-wrapper that it was "sticky" and then scroll wrapper would try to basically just do nothing. Easier to just not render it.